Update Flomesh implementation details (#2385)
kubernetes-sigs/gateway-api: site-src/implementations.md

### Flomesh Service Mesh (FSM)

> [Flomesh Service Mesh][fsm] is a community driven lightweight service mesh for Kubernetes East-West and North-South traffic management, uses ebpf for layer4 and pipy proxy for layer7 traffic management, support multi cluster network and comes bundled with an implementation of Ingress controller, Gateway API, Load Balancer, and cross-cluster service registration and service discovery.

Reviewer: This suggestion is just meant to break some stuff out a bit, feel free to dismiss this comment if it doesn't suit you. It also adds some links to some referenced technologies employed as a part of it.

```suggestion
[Flomesh Service Mesh][fsm] is a community driven lightweight service mesh for Kubernetes East-West and North-South traffic management. Flomesh uses [ebpf](https://www.kernel.org/doc/html/latest/bpf/index.html) for layer4 and [pipy](https://flomesh.io/pipy) proxy for layer7 traffic management. Flomesh comes bundled with a load balancer, cross-cluster service registration/discovery and it supports multi-cluster networking. It supports `Ingress` (and as such is an "Ingress controller") and Gateway API.
```
